data_structures/binary_tree/symmetric_tree.py

```diff
@@ -0,0 +1,95 @@
+"""
+Given the root of a binary tree, check whether it is a mirror of itself
+(i.e., symmetric around its center).
+
+Leetcode reference: https://leetcode.com/problems/symmetric-tree/
+"""
+
+
+class Node:
+    """
+    A Node has data variable and pointers to Nodes to its left and right.
+    """
+
+    def __init__(self, data: int) -> None:
+        self.data = data
+        self.left: Node | None = None
+        self.right: Node | None = None
+
+
+class SymmetricTree:
+    def is_symmetric_tree(self, tree: Node) -> bool:
+        r"""
+        Test cases for is_symmetric_tree function.
+
+        The below tree looks like this
+               1
+             /   \
+            2      2
+           / \    / \
+          3   4   4  3
+
+        >>> tree = Node(1)
+        >>> tree.left = Node(2)
+        >>> tree.right = Node(2)
+        >>> tree.left.left = Node(3)
+        >>> tree.left.right = Node(4)
+        >>> tree.right.left = Node(4)
+        >>> tree.right.right = Node(3)
```

Comment:
Please move these lines into a function defined as `def make_symmetric_tree() -> Node:`.

Reply:
there is no function named as  make_symmetric_tree(), are you expecting to create a new function in the name of  make_symmetric_tree(), can you please clarify

Reply:
Yes.  Please create the function.

Reply:
The reason that we want separate functions is that building the trees takes a lot of space and complexity and it is not really where we want visitors to focus their attention.  Also, by creating separate tree-building functions we can reuse them in multiple tests.